### ISSUE TITLE
Using equality comparison in selectors

### ISSUE THREAD

**Issue by Hoodrobinrs** [keajs/kea#143]
Hi, first of all, love the project, it's exactly what i needed in my project :)
I have a few questions regarding it, but I'm going to split those questions in separate issues to keep it clean.

My first question is about selectors, is it possible to use some kind of equality comparison in them? I could not find anything in the docs about it. Both [reselect](https://github.com/reduxjs/reselect#createselectorinputselectors--inputselectors-resultfunc-selectoroptions), which I think that Kea uses in the background and [react-redux](https://react-redux.js.org/api/hooks#equality-comparisons-and-updates) which I'm migrating from, have such features.

An example, that could benefit from such a feature:

```ts
import { kea } from 'kea';

import type { someLogicType } from './someLogicType';

type WidgetData = {}; 

export const someLogic = kea<someLogicType<WidgetData>>({
  actions: {
    addNewWidget: (widgetKey: string, data: WidgetData) => ({
      widgetKey,
      data,
    }),
  },
  reducers: {
    widgets: [
      {} as { [key: string]: WidgetData }, 
      {
        addNewWidget: (state, { widgetKey, data }) => {
          return { ...state, [widgetKey]: data };
        },
      },
    ],
  },
  selectors: {
    widgetKeys: [
      (selectors) => [selectors.widgets],
      (widgets) => {
        return Object.keys(widgets);
      },
    ],
  },
});

```

To explain what is going on here, the widgets reducer defines an object, with unique keys, each describing one widget, and in the value is some data, that the rendered widget uses. In the selector "widgetKeys" i'm returning all widget keys, which are used to render all of those widgets later in react. Currently, if any widget data changes, the widget that uses the "widgetKeys" selector rerenders, which is undesirable. 

It's also possible that my state structure is wrong. If there is a pattern that would fix my issue, without the need for equality comparisons, I would be really grateful for pointing me in the right direction. Thanks in advance and see you in the next issue :)

**Comment by mariusandra** [comment]
Hey, [version `2.5.11`](https://github.com/keajs/kea/compare/v2.5.10...v2.5.11) is out with this feature.

Now you can do:

```ts
import deepEqual from 'fast-deep-equal'

const logic = kea({
  selectors: {
    widgetKeys: [
      (selectors) => [selectors.widgets],
      (widgets) => {
        return Object.keys(widgets);
      },
      null, // PropTypes
      deepEqual
    ],
  },
```

The third element, `PropTypes`, is a legacy feature from the olden days, whose removal would be a breaking change. I'll do it in [version 3.0](https://v3.keajs.org/), which is actually just around the corner...

**Comment by mariusandra** [comment]
and sorry for the wait 😅

**Comment by mariusandra** [comment]
Oh, I didn't know of this new reselect 4.1 feature. 

I just released 2.6, and changed the last array element to just directly takes reselect's [`memoizeOptions`](https://github.com/reduxjs/reselect#defaultmemoizefunc-equalitycheckoroptions--defaultequalitycheck). You can do this now:

```js
const logic = kea({
    selectors: {
        widgetKeys: [
            (selectors) => [selectors.widgets],
            (widgets) => Object.keys(widgets),
            null, // PropTypes, will be removed in Kea 3.0
            { resultEqualityCheck: deepEqual },
        ],
    },
})
```
